File: CucumberJVM/Chapter13/13.1-LocalScenarioContextPattern/After/src/test/java/com/wimp/specs/stepdefinitions/AuthenticationStepDefinitions.java
package com.wimp.specs.stepdefinitions;

import com.wimp.app.services.AuthenticationService;
import io.cucumber.java.en.Given;

public class AuthenticationStepDefinitions {
    @Given("the customer {word} is authenticated")
    public void theCustomerIsAuthenticated(String customerName) {
        AuthenticationService.login(customerName);
    }
}

File: Reqnroll/Chapter13/13.2-ShareableScenarioContextPattern/Before/WIMP.Specs/StepDefinitions/AuthenticationStepDefinitions.cs
using Reqnroll;

using WIMP.App.Services;

namespace WIMP.Specs.StepDefinitions;

[Binding]
public class AuthenticationStepDefinitions
{
    [Given("the customer {string} is authenticated")]
    public void GivenTheCustomerIsAuthenticated(string customerName)
    {
        AuthenticationService.Login(customerName);
    }
}
